src/fdtables/commonconstants.rs

```diff
@@ -0,0 +1,117 @@
+pub const FDT_KINDEPOLL: u32 = 0xff00_0002;
+
+// These are the values we look up with at the end...
+// #[doc = include_str!("../docs/fdtableentry.md")]
+#[derive(Clone, Copy, Hash, Debug, PartialEq, Eq)]
+/// This is a table entry, looked up by virtual fd.
+pub struct FDTableEntry {
+    /// This is the kind of fd which it is.  These are user defined values
+    /// so they can track what this means.  Appropriate for storing information
+    /// which is global to all virtual fds that track this.
+    /// values over FDKINDMAX are reserved.
+    pub fdkind: u32,
+    /// underlying fd (could be the real, kernel fd below us or could be
+    /// some indicator of what table entry a virtual fd has.  It is up to
+    /// the implementer to decide how to use this.
+    pub underfd: u64,
+    /// Should I close this on exec?  Needed so fdtabls can implement
+    /// [/`empty_fds_for_exec`]
+    pub should_cloexec: bool,
+    /// Used to store fd specific extra information, such as flags or similar
+    /// which may differ for different 'dup'ed copies of a fd.   Whatever
+    /// the user desires may be placed here.
+    pub perfdinfo: u64,
+}
+
+#[allow(non_snake_case)]
+/// A function used when registering close handlers which does nothing...
+/// It is the default if no close handlers are defined
+pub const fn NULL_FUNC(_: FDTableEntry, _: u64) {}
+
+// BUG / TODO: Use this in some sane way...
+#[allow(dead_code)]
+/// Global maximum number of fds... (checks may not be implemented)
+pub const TOTAL_FD_MAX: u64 = 4096;
+
+// replicating these constants here so this can compile on systems other than
+// Linux...  Copied from Rust's libc.
+/// copied from libc
+pub const EPOLL_CTL_ADD: i32 = 1;
+/// copied from libc
+pub const EPOLL_CTL_MOD: i32 = 2;
+/// copied from libc
+pub const EPOLL_CTL_DEL: i32 = 3;
+
+#[allow(non_camel_case_types)]
+/// i32 copied from libc.  used in EPOLL event flags even though events are u32
+pub type c_int = i32;
+
+/// copied from libc
+pub const EPOLLIN: c_int = 0x1;
+/// copied from libc
+pub const EPOLLPRI: c_int = 0x2;
+/// copied from libc
+pub const EPOLLOUT: c_int = 0x4;
+/// copied from libc
+pub const EPOLLERR: c_int = 0x8;
+/// copied from libc
+pub const EPOLLHUP: c_int = 0x10;
+/// copied from libc
+pub const EPOLLRDNORM: c_int = 0x40;
+/// copied from libc
+pub const EPOLLRDBAND: c_int = 0x80;
+/// copied from libc
+pub const EPOLLWRNORM: c_int = 0x100;
+/// copied from libc
+pub const EPOLLWRBAND: c_int = 0x200;
+/// copied from libc
+pub const EPOLLMSG: c_int = 0x400;
+/// copied from libc
+pub const EPOLLRDHUP: c_int = 0x2000;
+/// copied from libc
+pub const EPOLLEXCLUSIVE: c_int = 0x1000_0000;
+/// copied from libc
+pub const EPOLLWAKEUP: c_int = 0x2000_0000;
+/// copied from libc
+pub const EPOLLONESHOT: c_int = 0x4000_0000;
+// Turning this on here because we copied from Rust's libc and I assume they
+// intended this...
+#[allow(overflowing_literals)]
```

Comment:
It might cause issues. If possible, please add a comment explaining why it is required.

Reply:
```
// Turning this on here because we copied from Rust's libc and I assume they
// intended this...
```
This explains why we use `#[allow(overflowing_literals)]` here.

To be more detailed:

`0x8000_0000` is a hexadecimal literal representing a 32-bit integer value in rust and works for unsigned int. But it will cause warning for signed int (ie `c_int`)

Since the value is copied from Rust's libc and is meant to match certain system constants (such as EPOLLET), this case is by design.
Use #[allow(overflowing_literals)] to ensure our code compiles without warnings.

Reply:
Great question, added more detailed comments